tests/test_Model.py

```diff
@@ -0,0 +1,181 @@
+import pandas as pd
+import pytest
+import xgboost
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score
+
+import modelgym
+from modelgym.trainer import Trainer
+from modelgym.util import TASK_CLASSIFICATION
+from modelgym.util import split_and_preprocess
+
+TEST_SIZE = 0.2
+N_CV_SPLITS = 2
+NROWS = 1000
+N_PROBES = 2
+N_ESTIMATORS = 100
+TEST_PARAMS = ["classification", "range", "regression"]
+APPROVED_PARAMS = ["classification", "regression"]
+MODEL_CLASS = [modelgym.XGBModel, modelgym.LGBModel, modelgym.RFModel]
+
+
+def test_preprocess_params():
+    for par1 in TEST_PARAMS:
+        # assert APPROVED_PARAMS.__contains__(par1)
+        try:
+            model = modelgym.XGBModel(learning_task=par1)
+        except ValueError:
+            print(par1, "not expected")
+        k = len(model.default_params)
+        model.preprocess_params(model.default_params)
+        assert TEST_PARAMS.__contains__(model.learning_task)
+        assert len(model.default_params) != k + 3
+    return 0
+
+
+@pytest.mark.xfail
```

Comment:
remove xfail, and inverse last condition in assert (using @pytest.mark.xfail with a check function is probably better for something like documenting unfixed bugs)